### PR TITLE
[MRG] Refactor subpackages

### LINKED ISSUE #545: Refactor subpackage structure
* `phy.cluster`
* `phy.cluster.algorithms`
* `phy.detect`
* `phy.session`

### LINKED ISSUE #549: Refactor default settings
* Every subpackage might have a `default_settings.py` file with default settings related to the functionality it implements
* There is a function `load_default_settings(paths)` that takes as input a list of paths (like `['phy/cluster/algorithms', 'phy/cluster']`), and returns a dictionary with the settings loaded from the `default_settings.py` files, in the requested order

### PR DESCRIPTION
**Pull request by rossant**
* Closes #545 
* Closes #549

### CODE REVIEW

**Comment by rossant** [comment]
ready to merge @nippoo 
we may tweak the directory structure further later. First I have a few bugs to fix...

File: phy/cluster/manual/tests/test_view_models.py [hunk 1]
```diff
@@ -10,14 +10,14 @@
 
 from pytest import mark
 
-from ...utils.array import _spikes_per_cluster
-from ...utils.logging import set_level
-from ...utils.testing import (show_test_start,
+from ....utils.array import _spikes_per_cluster
```
Comment (nippoo) [inline phy/cluster/manual/tests/test_view_models.py]: lol

File: phy/utils/settings.py [hunk 2]
```diff
@@ -51,40 +90,35 @@ def _update(self, d):
     def _try_load_json(self, path):
         try:
             self._update(_load_json(path))
-            debug("Loaded internal settings file "
-                  "from `{}`.".format(path))
+            # debug("Loaded internal settings file "
```
Comment (nippoo) [inline phy/utils/settings.py]: why commented?

Comment (rossant) [inline phy/utils/settings.py]: it tends to clutter the test suite a lot...

**Pull request merged**